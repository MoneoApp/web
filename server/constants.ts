export const secret = process.env.SECRET ?? '';
export const reflection = Boolean(process.env.REFLECTION);
