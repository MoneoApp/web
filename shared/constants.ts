export enum Error {
  Unknown = 'UNKNOWN',
  Unauthorized = 'UNAUTHORIZED',
  BadUserInput = 'BAD_USER_INPUT',
  InvalidInvite = 'INVALID_INVITE',
  InvalidFileType = 'INVALID_FILE_TYPE',
  InvalidInteractions = 'INVALID_INTERACTIONS'
}
