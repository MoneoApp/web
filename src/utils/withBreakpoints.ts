import { css, SerializedStyles } from '@emotion/react';

import { Breakpoint, Breakpoints } from '../types';

import { withBreakpoint } from './withBreakpoint';

export function withBreakpoints<T>(
  breakpoints: Breakpoints<T>,
  getStyle: (value: T, breakpoint: Breakpoint) => SerializedStyles
) {
  let result: SerializedStyles | undefined;

  for (const key in breakpoints) {
    if (!breakpoints.hasOwnProperty(key)) {
      continue;
    }

    const br = key as Breakpoint;
    const value = breakpoints[br];

    if (value !== undefined) {
      result = css`
        ${result};
        ${withBreakpoint(br, getStyle(value, br))};
      `;
    }
  }

  return result;
}
