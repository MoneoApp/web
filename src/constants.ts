import { faCircle, faSquare, IconDefinition } from '@fortawesome/free-solid-svg-icons';
import { ComponentType } from 'react';

import { Error } from '../shared/constants';

import { InteractionType, UserRole } from './apollo/globalTypes';
import { CircleShape } from './components/editor/shapes/CircleShape';
import { SquareShape } from './components/editor/shapes/SquareShape';
import { Breakpoint, Color, Shade, ShapeConfig } from './types';

export const breakpoints: Record<Breakpoint, number> = {
  phone: 0,
  tabletPortrait: 37.5,
  tabletLandscape: 56.25,
  laptop: 75,
  desktop: 93.75
};

export const colors: Record<Color, Partial<Record<Shade, [string, string]>>> = {
  gray: {
    0: ['#ffffff', '#1f1f1f'],
    100: ['#efefef', '#303030'],
    200: ['#d4d4d4', '#404040'],
    300: ['#9a9a9a', '#5f5f5f'],
    400: ['#707070', '#929292'],
    500: ['#454545', '#efefef']
  },
  yellow: {
    100: ['#fdecbe', '#906f18'],
    200: ['#ffba05', '#ffba05'],
    300: ['#d89b00', '#ffdd82']
  },
  green: {
    100: ['#9eddc2', '#167b4f'],
    200: ['#27b376', '#27b376'],
    300: ['#167b4f', '#9eddc2']
  },
  red: {
    100: ['#faa7ae', '#bf212f'],
    200: ['#e5414f', '#e5414f'],
    300: ['#bf212f', '#faa7ae']
  }
};

export const messages: Record<Error, string> = {
  [Error.Unknown]: 'Oops, er is iets mis gegaan',
  [Error.Unauthorized]: 'Je mag deze actie niet uitvoeren',
  [Error.BadUserInput]: 'De ingevulde gegevens zijn ongeldig',
  [Error.InvalidInvite]: 'De uitnodiging is ongeldig',
  [Error.InvalidFileType]: 'Ongeldig bestandstype'
};

export const errors: Record<string, string> = {
  any: 'Ongeldige invoer',
  string: 'Ongeldige invoer',
  email: 'Ongeldig e-mailadres',
  password: 'Ongeldig wachtwoord',
  invalidCredentials: 'Ongeldige inloggegevens',
  emailInUse: 'E-mailadres al in gebruik'
};

export const roles: Record<UserRole, string> = {
  USER: 'Gebruiker',
  ADMIN: 'Beheerder'
};

export const shapes: Record<InteractionType, ComponentType<ShapeConfig>> = {
  [InteractionType.SQUARE]: SquareShape,
  [InteractionType.CIRCLE]: CircleShape
};

export const shapeIcons: Record<InteractionType, IconDefinition> = {
  [InteractionType.SQUARE]: faSquare,
  [InteractionType.CIRCLE]: faCircle
};
