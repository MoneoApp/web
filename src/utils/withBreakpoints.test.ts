import { css } from '@emotion/react';

import { breakpoints } from '../constants';

import { withBreakpoints } from './withBreakpoints';

it('should insert the correct breakpoints', () => {
  expect(withBreakpoints(breakpoints, (value) => css`
    content: "${value}";
  `)?.styles).toMatchSnapshot();
});
