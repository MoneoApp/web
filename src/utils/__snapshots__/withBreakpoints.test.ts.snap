// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should insert the correct breakpoints 1`] = `";content:\\"0\\";;;;label:result;;;@media (min-width: 37.5rem){content:\\"37.5\\";;;};label:withBreakpoint;;;;label:result;;;@media (min-width: 56.25rem){content:\\"56.25\\";;;};label:withBreakpoint;;;;label:result;;;@media (min-width: 75rem){content:\\"75\\";;;};label:withBreakpoint;;;;label:result;;;@media (min-width: 93.75rem){content:\\"93.75\\";;;};label:withBreakpoint;;;;label:result;"`;
